src/Components/Components/src/PublicAPI.Unshipped.txt

```diff
@@ -40,7 +40,7 @@ Microsoft.AspNetCore.Components.NavigationOptions.ForceLoad.init -> void
 Microsoft.AspNetCore.Components.NavigationOptions.NavigationOptions() -> void
 Microsoft.AspNetCore.Components.NavigationOptions.ReplaceHistoryEntry.get -> bool
 Microsoft.AspNetCore.Components.NavigationOptions.ReplaceHistoryEntry.init -> void
-Microsoft.AspNetCore.Components.RenderHandle.IsHotReloading.get -> bool
+Microsoft.AspNetCore.Components.RenderHandle.IsRenderingOnMetadataUpdate.get -> bool
```

Comment:
The runtime is using `MetadataUpdate` as the term for hot reload. Updated this API to use this nomenclature. Outside of this, this PR is fairly simple - it uses the new API and adds a test hook for code that currently relies on testing. Once the SDK change - https://github.com/dotnet/sdk/pull/18810 lands, we're able to trim out all[1] references to hot reload specific code.

[1] Right now the test type with the property set to default ends up not being trimmed. But it seems like a small cost and I'll follow up to see if we can clear that up. I wouldn't block on it though.